src/video/wayland/SDL_waylandwindow.c

```diff
@@ -1140,25 +1369,64 @@ void Wayland_SetWindowSize(_THIS, SDL_Window * window)
     SDL_VideoData *data = _this->driverdata;
     SDL_WindowData *wind = window->driverdata;
     struct wl_region *region;
+#ifdef HAVE_LIBDECOR_H
+    struct libdecor_state *state;
+#endif
 
     wl_surface_set_buffer_scale(wind->surface, get_window_scale_factor(window));
 
     if (wind->egl_window) {
         WAYLAND_wl_egl_window_resize(wind->egl_window, window->w * get_window_scale_factor(window), window->h * get_window_scale_factor(window), 0, 0);
     }
 
+#ifdef HAVE_LIBDECOR_H
+    if (data->shell.libdecor) {
+        state = libdecor_state_new(window->w, window->h);
```

Comment:
Needs a null check for wind->shell_surface.libdecor.frame, once it's shown it should be the correct size...?